tests/features/user/register.feature

```diff
@@ -60,3 +60,27 @@ Feature: User registration
       | subject   | Your Joinup account was created successfully.                                                                                   |
       | body      | The Joinup Support Team created your account on Joinup. Please log in through the following link in order to set your password. |
     And the account for SuperUser should be active
+
+  Scenario: A user account whose first and last names are the identical is
+    deleted after creation but the user is receiving all notification as if the
+    account would have been created.
+
+    Given I am an anonymous user
+    And I go to "/user/register"
+    When I fill in "E-mail" with "spam@example.com"
+    And I fill in "Username" with "spam"
+    And I fill in "First name" with "Spam"
+    And I fill in "Family name" with "Spam"
+    Given I press "Create new account"
+    Then I should see the success message "Thank you for applying for an account. Your account is currently pending approval by the site administrator."
+    And the following email should have been sent:
+      | recipient_mail | spam@example.com                                                                                                                                                                                                                                 |
+      | subject        | Your Joinup account is pending approval.                                                                                                                                                                                                         |
+      | body           | Thank you for registering at Joinup. Your application for an account is currently pending approval. Once it has been approved, you will receive another email containing information about how to sign in, set your password, and other details. |
+    # Creating a new account with same username and E-mail just to prove that,
+    # the in previous attempt, the user has been deleted.
+    Given user:
```

Comment:
Let's create a step similar to `\FeatureContext::assertUserExistence()` that tests that a user is not there.
So we can keep our business language and say `And I should not have a "spam" user`.